### PR TITLE
Feat/pex filters integration

### PR DESCRIPTION
**Pull request by maycon-mello**

### CODE REVIEW

File: src/features/credentials/credentials.js [hunk 1]
```diff
@@ -91,19 +91,37 @@ export const validateCredential = credential => {
...
   const {credentials, saveCredential, deleteCredential} = useCredentialUtils();
   const [items, setItems] = useState([]);
   const [refreshing, setRefreshing] = useState(false);
 
   const syncCredentials = useCallback(async () => {
     setRefreshing(true);
+    const filteredCredentials = presentationDefinition
+      ? (
+          await credentialServiceRPC.filterCredentials({
+            credentials: credentials.map(item => item.content),
```
Comment (mike-parkhill) [inline src/features/credentials/credentials.js:109]: could `credentials` be undefined and/or the individual `item`? i.e. do we need some checks in here?

File: src/features/credentials/credentials.js [hunk 2]
```diff
@@ -91,19 +91,37 @@ export const validateCredential = credential => {
...
+            presentationDefinition,
+            holderDid,
+          })
+        ).verifiableCredential.map(item =>
+          credentials.find(vc => vc.content.id === item.id),
```
Comment (mike-parkhill) [inline src/features/credentials/credentials.js:114]: same here, could `item` be undefined or not an object?

File: src/features/verifier/CredentialVerifier.js [hunk 3]
```diff
@@ -102,52 +104,79 @@ function CredentialListItem({credential}) {
...
 }
 
-function getErrorMessage({result} = {}) {
-  if (!result) {
-    return null;
-  }
+const verifyPresentation = async (presentation, presentationDefinition) => {
+  const credentialCheck = await Promise.all(
+    presentation.verifiableCredential.map(credential =>
+      credentialServiceRPC.verifyCredential({
```
Comment (mike-parkhill) [inline src/features/verifier/CredentialVerifier.js:110]: skip if `credential` is undefined?

File: src/features/verifier/CredentialVerifier.js [hunk 4]
```diff
@@ -102,52 +104,79 @@ function CredentialListItem({credential}) {
...
+      }),
+    ),
+  );
+
+  let isValid = credentialCheck.reduce(
+    (value, result) => result.verified && value,
+    true,
+  );
+
+  console.log(credentialCheck);
```
Comment (mike-parkhill) [inline src/features/verifier/CredentialVerifier.js:121]: are these logs staying?

File: src/features/verifier/CredentialVerifier.js [hunk 5]
```diff
@@ -102,52 +104,79 @@ function CredentialListItem({credential}) {
...
-    return result?.error;
+    console.log(result);
+
+    isValid = !result.errors.length;
   }
 
-  if (result && !result.verified) {
-    return result.results?.[0]?.error?.message && result.error;
+  if (!isValid) {
+    throw new Error('Invalid presentation');
```
Comment (mike-parkhill) [inline src/features/verifier/CredentialVerifier.js:136]: any context we can provide here? i.e. why is the presentation invalid?

File: src/features/verifier/CredentialVerifier.js [hunk 6]
```diff
@@ -218,6 +236,10 @@ export const CredentialVerifier = ({
 
 export const CredentialVerifierContainer = withErrorBoundary(({route}) => {
   const params = route.params || {};
+  const {documents} = useWallet({syncDocs: true});
+  const templates = documents.filter(
+    doc => doc.type === 'VerificationRequestTemplate',
```
Comment (mike-parkhill) [inline src/features/verifier/CredentialVerifier.js:241]: could doc ever be undefined?

File: src/features/verifier/VerificationRequestTemplateScreen.js [hunk 7]
```diff
@@ -138,23 +149,51 @@ export function VerificationRequestTemplateContainer({}) {
   );
 
   const importTemplate = data => {
+    const definition = data.presentation_definition;
+
+    if (!definition) {
+      throw new Error('Invalid template');
```
Comment (mike-parkhill) [inline src/features/verifier/VerificationRequestTemplateScreen.js:155]: message should probably be more like, "Template not found" or something

File: src/features/credentials/credentials.js (hunk 1 above)
Comment (maycon-mello) [reply src/features/credentials/credentials.js]: Sure, fixed

File: src/features/credentials/credentials.js (hunk 2 above)
Comment (maycon-mello) [reply src/features/credentials/credentials.js]: Sure, fixed

File: src/features/verifier/CredentialVerifier.js (hunk 5 above)
Comment (maycon-mello) [reply src/features/verifier/CredentialVerifier.js:135]: Still need to figure out how to extract detailed errors from constraints. I propose we handle this in a separate ticket, including a good UX for that.

Comment (mike-parkhill) [reply src/features/verifier/CredentialVerifier.js:135]: Ok, can you add that ticket along with whatever thoughts/details you have?

**Review by mike-parkhill — APPROVED** [review]

File: src/features/verifier/CredentialVerifier.js [hunk 8]
```diff
@@ -102,52 +104,79 @@ function CredentialListItem({credential}) {
...
     setLoading(true);
-    setError(false);
+    setError(null);
     setResult(null);
 
-    credentialServiceRPC
-      .verifyCredential({credential: presentation.verifiableCredential[0]})
-      .then(setResult)
-      .catch(setError)
+    verifyPresentation(presentation, presentationDefinition)
```
Comment (SamHellawell) [inline src/features/verifier/CredentialVerifier.js:156]: all the naming is "credential verifier" but i dont see this verifying credentials anymore, just presentations?

**Pull request merged**